### PR TITLE
image-info: various cleanups and use loop devices instead of nbd

### PR DESCRIPTION
**Pull request by gicmo**
I am preparing to remove the usage of `nbd` from `image-info` but wanted to give the code a bit of a cleanup before I started doing that. See the individual commit messages for details.

### CODE REVIEW

**Comment by ondrejbudai** [comment]
Changes looking good to me. However, they uncovered the raciness of nbd in its full potential resulting in 100% failure rate in Travis. Let's put the `nbd` removal also in this PR, so we don't have to merge code which is making our CI really unhappy.

File: cmd/osbuild-image-tests/constants/constants-travis.go [hunk 1]
```diff
@@ -18,13 +21,23 @@ func GetOsbuildCommand(store, outputDirectory string) *exec.Cmd {
...
+}
+
 var TestPaths = struct {
 	ImageInfo               string
 	PrivateKey              string
 	TestCasesDirectory      string
 	UserData                string
 	MetaData                string
 	AzureDeploymentTemplate string
+	PythonPath              string
```
Comment (gicmo) [inline cmd/osbuild-image-tests/constants/constants-travis.go:40]: Fixed locally.

**Comment by gicmo** [comment]
If the way to set the `PYTHONPATH` on travis is ok I will move the commit to the very top, so that when the actual change gets introduce it does not break that commit.

File: cmd/osbuild-image-tests/constants/constants-travis.go (hunk 1 above)
Comment (ondrejbudai) [reply cmd/osbuild-image-tests/constants/constants-travis.go:40]: LGTM!

**Comment by ondrejbudai** [comment]
The only failure is because of iot, see #798

**Pull request merged**